flake.nix

```diff
@@ -0,0 +1,50 @@
+{
+  epoch = 201909;
+
+  description = "A collection of packages for the Nix package manager";
+
+  outputs = { self }:
+    let
+      pkgs = import ./. { system = "x86_64-linux"; };
+      jobs = import ./pkgs/top-level/release.nix {
+        nixpkgs = self;
+      };
+      lib = import ./lib;
+    in
+    {
+      lib = lib // {
+        nixosSystem = { modules, ... } @ args:
+          import ./nixos/lib/eval-config.nix (args // {
+            modules = modules ++
+              [ { system.nixos.versionSuffix =
+                    ".${lib.substring 0 8 self.lastModified}.${self.shortRev}";
+                  system.nixos.revision = self.rev;
+                }
+              ];
+          });
+      };
+
+      checks.tarball = jobs.tarball;
+
+      builders = {
+        inherit (pkgs) stdenv fetchurl;
+      };
+
+      htmlDocs = {
+        nixpkgsManual = jobs.manual;
+        nixosManual = (import ./nixos/release-small.nix {
+          nixpkgs = self;
+        }).nixos.manual.x86_64-linux;
+      };
+
+      packages = {
+        inherit (pkgs) hello nix fuse nlohmann_json boost firefox;
+      };
+
+      legacyPackages = pkgs;
+
+      nixosModules = {
+        notDetected = ./nixos/modules/installer/scan/not-detected.nix;
```

Comment:
Why include this?

Reply:
> Note: NIX_PATH is unavailable in flake-based builds.

---

Should we add other modules as well? For example, I use the following images on my server:

    <nixpkgs/nixos/modules/profiles/minimal.nix>
    <nixpkgs/nixos/modules/virtualisation/container-config.nix>
    <nixpkgs/nixos/modules/installer/cd-dvd/channel.nix>